### ISSUE TITLE
Problem with Koboclud and Kobo Forma

### ISSUE THREAD

**Issue by DNDaniele** [fsantini/KoboCloud#119]
I have a problem on my Kobo Forma, basically it offers me the possibility of synchronization with dropbox, but since all my archive is on google drive I have well thought of installing the Kobocloud plugin.

As instructions I install it, configure it with my shared folder, wait a minute and he downloads all my ebooks correctly!
I leave it for a few hours and resume it to start using it, I try to open a book but I receive an error (that I am not authorized to open it) and shortly after I find my library totally empty, I connect it to the PC and I notice that the ebooks are present under 'Library' but are not displayed.

I also own a Kobo Glo but I have never had any problems on this, both the Glo and the Forma are updated to the latest version (4.31.19086). I also tried resetting the device but again it happened. Now the library on the Forma is empty (in my books), while on the Glo it continues to work correctly!

Can you give me an idea on how to try to solve this?
Thank you

**Comment by nsainaney** [comment]
I wonder if the issue is related to the one I'm facing here: https://www.mobileread.com/forums/showthread.php?p=4255323.

@fsantini, could the following line be device dependent?

`echo sd add /dev/mmcblk1p1 >> /tmp/nickel-hardware-status`

**Comment by fsantini** [comment]
I don't think so, but I don't have any way to test this unfortunately. It looks like a directory permission problem. One should navigate in the ereader file system and see the permissions of the downloaded files.

**Comment by nsainaney** [comment]
Good idea. I'll check that.

In the meantime, I have been experimenting with NickelDBus. The following works like a charm to refresh the library:

`qndb -t 30000 -s pfmDoneProcessing -m pfmRescanBooksFull`

**Comment by fsantini** [comment]
Fantastic! I'll try it out and hopefully we can put it in the codebase. Thanks!
